build.sbt

```diff
@@ -1,124 +1,228 @@
+    new RuleTransformer(new RewriteRule {
+      override def transform(node: xml.Node): Seq[xml.Node] = node match {
+        case e: Elem
+          if e.label == "dependency" && e.child.exists(child => child.label == "groupId" && child.text == "org.scoverage") => Nil
+        case _ => Seq(node)
+      }
+    }).transform(node).head
+  },
+
+  pomExtra in ThisBuild :=
+    <url>https://github.com/monix/shade</url>
+    <licenses>
+      <license>
+        <name>The MIT License</name>
+        <url>http://opensource.org/licenses/MIT</url>
+        <distribution>repo</distribution>
+      </license>
+    </licenses>
+    <scm>
+      <url>git@github.com:monix/shade.git</url>
+      <connection>scm:git:git@github.com:monix/shade.git</connection>
+    </scm>
+    <developers>
+      <developer>
+        <id>alex_ndc</id>
+        <name>Alexandru Nedelcu</name>
+        <url>https://alexn.org</url>
+      </developer>
+    </developers>
 )
 
-libraryDependencies += ("org.scala-lang" % "scala-reflect" % scalaVersion.value % "compile")
-
-// -- Settings meant for deployment on oss.sonatype.org
+lazy val cmdlineProfile =
+  sys.props.getOrElse("sbt.profile", default = "")
 
-useGpg := true
-useGpgAgent := true
-usePgpKeyHex("2673B174C4071B0E")
+def profile: Project ⇒ Project = pr => cmdlineProfile match {
+  case "coverage" => pr
+  case _ => pr.disablePlugins(scoverage.ScoverageSbtPlugin)
+}
 
-publishMavenStyle := true
+lazy val doNotPublishArtifact = Seq(
+  publishArtifact := false,
+  publishArtifact in (Compile, packageDoc) := false,
+  publishArtifact in (Compile, packageSrc) := false,
+  publishArtifact in (Compile, packageBin) := false
+)
 
-publishTo := {
-  val nexus = "https://oss.sonatype.org/"
-  if (isSnapshot.value)
-    Some("snapshots" at nexus + "content/repositories/snapshots")
-  else
-    Some("releases"  at nexus + "service/local/staging/deploy/maven2")
-}
+lazy val crossSettings = sharedSettings ++ Seq(
+  unmanagedSourceDirectories in Compile += {
+    baseDirectory.value.getParentFile / "shared" / "src" / "main" / "scala"
+  },
+  unmanagedSourceDirectories in Test += {
+    baseDirectory.value.getParentFile / "shared" / "src" / "test" / "scala"
+  }
+)
 
-publishArtifact in Test := false
-pomIncludeRepository := { _ => false } // removes optional dependencies
-
-scalariformSettings
-
-pomExtra in ThisBuild :=
-  <url>https://github.com/monix/shade</url>
-  <licenses>
-    <license>
-      <name>The MIT License</name>
-      <url>http://opensource.org/licenses/MIT</url>
-      <distribution>repo</distribution>
-    </license>
-  </licenses>
-  <scm>
-    <url>git@github.com:monix/shade.git</url>
-    <connection>scm:git:git@github.com:monix/shade.git</connection>
-  </scm>
-  <developers>
-    <developer>
-      <id>alex_ndc</id>
-      <name>Alexandru Nedelcu</name>
-      <url>https://alexn.org</url>
-    </developer>
-  </developers>
+lazy val scalaJSSettings = Seq(
+  coverageExcludedFiles := ".*"
```

Comment:
This isn't about Minitest. I've made `shade-local` to cross-compile to Scala.js, since there's nothing JVM specific in it and who knows, might be useful to somebody in the browser :-) 

The problem is that sbt-scoverage doesn't work with Scala.js. So we have to exclude all Scala.js files from coverage reports.